src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusMetric.cs

```diff
@@ -41,10 +42,31 @@ public PrometheusMetric(string name, string unit, PrometheusType type, bool disa
             // https://github.com/open-telemetry/opentelemetry-specification/blob/b2f923fb1650dde1f061507908b834035506a796/specification/compatibility/prometheus_and_openmetrics.md#L242-L246
             if (!sanitizedName.Contains(sanitizedUnit))
```

Comment:
```suggestion
            if (!sanitizedName.EndsWith(sanitizedUnit))
```

The current spec is intended to produce the same name for OpenMetrics and non-OpenMetrics.  Rather than have different logic for OM vs non-OM, lets update the spec to require the unit to be a suffix (before type suffixes)

Reply:
https://github.com/open-telemetry/opentelemetry-specification/pull/4057